code/modules/power/supermatter/supermatter.dm

```diff
@@ -535,19 +555,23 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 			air_update_turf()
 
 	//Makes em go mad and accumulate rads.
+	var/toAdd = -0.05
 	for(var/mob/living/carbon/human/l in view(src, HALLUCINATION_RANGE(power))) // If they can see it without mesons on.  Bad on them.
-		if(!istype(l.glasses, /obj/item/clothing/glasses/meson))
+		if(l.job == "Psychologist")
```

Comment:
assigned role only

Reply:
```suggestion
		if(l.mind?.assigned_role == "Psychologist")
```

also god these aren't defined...

Reply:
I'm so sorry

Reply:
oh and I seem to recall changing this off assigned so people could engi job change as psychologist

Reply:
If the 2 choices are intrinsic and anyone can be given the role and heal the SM, I'd prefer it to be intrinsic.